iothub/device/samples/convention-based-samples/readme.md

```diff
@@ -12,41 +12,557 @@ products:
 urlFragment: azure-iot-pnp-device-samples-for-csharp-net
 ---
 
-# IoT Plug And Play device samples
+# IoT Plug And Play (PnP) device/ module APIs
 
-These samples demonstrate how a device that follows the [IoT Plug and Play conventions][pnp-convention] interacts with IoT Hub or IoT Central, to:
+Device(s)/ module(s) connecting to IoT Hub that announce their DTDL model Id during initialization can now perform convention-based operations. One such convention supported is [IoT Plug and Play][pnp-convention].
 
-- Send telemetry.
-- Update read-only and read-write properties.
-- Respond to command invocation.
+These device(s)/ module(s) can now use the native PnP APIs in the Azure IoT device SDKs to directly exchange messages with an IoT Hub, without having to manually format these messages to follow the PnP convention.
 
-The samples demonstrate two scenarios:
+## Table of Contents
 
-- An IoT Plug and Play device that implements the [Thermostat][d-thermostat] model. This model has a single interface that defines telemetry, read-only and read-write properties, and commands.
-- An IoT Plug and Play device that implements the [Temperature controller][d-temperature-controller] model. This model uses multiple components:
-  - The top-level interface defines telemetry, read-only property and commands.
-  - The model includes two [Thermostat][thermostat-model] components, and a [device information][d-device-info] component.
+- [Client initialization](#client-initialization)
+  - [Announce model Id during client initialization (same as in latest `master` release)](#announce-model-id-during-client-initialization-same-as-in-latest-master-release)
```

Comment:
I believe the convention for Id is ID in read me documents. I could be wrong though.

Reply:
this doc does say "ID", so I'll keep it consistent

Reply:
https://docs.microsoft.com/en-us/azure/iot-pnp/concepts-developer-guide-device?pivots=programming-language-csharp